authentik/events/models.py

```diff
@@ -160,6 +160,51 @@ def get_events_per(
         return self.get_queryset().get_events_per(time_since, extract, data_points)
 
 
+class EventBatch(models.Model):
+    """Model to store information about batches of events."""
+
+    batch_id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    event_type = models.CharField(max_length=255)
+    start_time = models.DateTimeField(auto_now_add=True)
+    end_time = models.DateTimeField(null=True, blank=True)
+    event_count = models.IntegerField(default=0)
+    last_updated = models.DateTimeField(auto_now=True)
+    max_batch_size = models.IntegerField(default=10)
+    batch_timeout = models.IntegerField(default=60)  # Timeout in seconds
+
+    @staticmethod
+    def get_or_create_batch(action):
+        """Get or create a batch for a given action."""
+        existing_batch = EventBatch.objects.filter(event_type=action, end_time__isnull=True).first()
+        if existing_batch:
+            return existing_batch
+        return EventBatch.objects.create(event_type=action)
```

Comment:
One thing I would add is the `app` field to group events by, and maybe also `user`

Reply:
Done!